lightning/src/ln/peer_handler.rs

```diff
@@ -617,7 +614,12 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref> PeerManager<D
 	pub fn write_buffer_space_avail(&self, descriptor: &mut Descriptor) -> Result<(), PeerHandleError> {
 		let mut peers = self.peers.lock().unwrap();
 		match peers.peers.get_mut(descriptor) {
-			None => panic!("Descriptor for write_event is not already known to PeerManager"),
+			None => {
+				// This is most likely a simple race condition where the user found that the socket
+				// was writeable, then we told the user to `disconnect_socket()`, then they called
+				// this method. Return an error to make sure we get disconnected.
+				return Err(PeerHandleError { no_connection_possible: false });
```

Comment:
I think the doc comment describing a panic above is outdated now (and possibly other parts of the comment that prescribe reentrancy caution?)

Reply:
It is removed in a later commit - is the status of it at HEAD ok?

Reply:
Oops, yup